test: Migrate request queuing batch-txs-per-dapp-extra-tx to Typescript and POM

Unik0rnMaggie opened the pull request:
## **Description**

Migrate the following e2e test spec file to Typescript and Page object modal:

`test/e2e/tests/request-queuing/batch-txs-per-dapp-extra-tx.spec.js`

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/34724?quickstart=1)

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and accurately)
-->

CHANGELOG entry:

## **Related issues**

Fixes: [MMQA-751](https://consensyssoftware.atlassian.net/browse/MMQA-751?atlOrigin=eyJpIjoiYjE1MTg0NmFkMjM0NDI5OGI0MzUzZDQwZjZkNTU0ZDAiLCJwIjoiaiJ9)

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.


[MMQA-751]: https://consensyssoftware.atlassian.net/browse/MMQA-751?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Unik0rnMaggie committed:
test: Migrate request queuing batch-txs-per-dapp-extra-tx to Typescript and POM

Unik0rnMaggie committed:
chore: updates following bot review comment

Unik0rnMaggie committed:
Merge branch 'main' into migrate-request-queuing-batch-txs-per-dapp-extra-tx-spec

Unik0rnMaggie committed:
Merge branch 'main' into migrate-request-queuing-batch-txs-per-dapp-extra-tx-spec

Unik0rnMaggie committed:
chore: updates following review comment

hjetpoluru commented on test/e2e/tests/request-queuing/batch-txs-per-dapp-extra-tx.spec.ts:
```diff
@@ -0,0 +1,141 @@
+        await driver.switchToWindowWithUrl(DAPP_URL);
+        await firstTestDapp.check_networkIsConnected('0x539');
+        await firstTestDapp.clickSimpleSendButton();
+        await firstTestDapp.clickSimpleSendButton();
```
Why do we need to call `clickSimpleSendButton `twice? 

Instead, we could use the `Send` button with the `clickElementAndWaitToDisappear `method, since in the next step we already have `waitUntilXWindowHandles`.

hjetpoluru replied:
Ok - this involves two transactions, so the confirmation button needs to be clicked twice and validation of the corresponding windows is required. 

Therefore my above review comment can be skipped and the same applies to the other comments as well

hjetpoluru commented on test/e2e/tests/request-queuing/batch-txs-per-dapp-extra-tx.spec.ts:
```diff
@@ -0,0 +1,141 @@
+        // Dapp 2 send 2 tx
+        await driver.switchToWindowWithUrl(DAPP_ONE_URL);
+        await secondTestDapp.check_networkIsConnected('0x53a');
+        await secondTestDapp.clickSimpleSendButton();
```
Similar observation as above.

hjetpoluru reviewed (dismissed):
LGTM

Unik0rnMaggie committed:
Merge branch 'main' into migrate-request-queuing-batch-txs-per-dapp-extra-tx-spec

Unik0rnMaggie committed:
chore: fix lint errors due to methods name updates

Unik0rnMaggie committed:
chore: finx lint error

Unik0rnMaggie committed:
Resolve merge conflict: keep batch-txs-per-dapp-extra-tx.spec.js from main

The file was deleted in HEAD but modified in main. Since this branch
appears to be working on migrating this specific test file, keeping
the latest version from main to continue the migration work.

Unik0rnMaggie committed:
Merge branch 'main' into migrate-request-queuing-batch-txs-per-dapp-extra-tx-spec

Unik0rnMaggie committed:
delete js file following merge conflict

Unik0rnMaggie committed:
Merge branch 'main' into migrate-request-queuing-batch-txs-per-dapp-extra-tx-spec

seaona reviewed (approved):
LGTM!
